### PR TITLE
Version 3.0.0

### LINKED ISSUE #5: Limitations of approach unusual for express middleware
Interface for adding these routes is unusual for express. Typically middlewares are attached via `app.use()` 

The current design creates some difficulties for the webapp:
- In the webapp we have our own middleware that attaches extra properties to the request. But this service integration doesn't expose the usual `res`/`req`/`next`, so we can't access our request properties and need to make them global.
- Our `__about` needs to get version, which is asynchronous (because in dev it calls `git describe`). We can't do that with literal opts.about. We also expose `req.secure` in __about for debugging. Without access to `req` we can't.

### PR DESCRIPTION
**Pull request by rowanmanning**
This work forms the basis of v3.0.0 switches to a middleware-based API. Aside from the rename from `express-ftwebservice` to `@financial-times/express-web-service`, the key differences are illustrated here:

```js
const express = require('express');
const expressWebService = require('@financial-times/express-web-service');
const app = express();
const options = {};

// Old
expressWebService(app, options);

// New
app.use(expressWebService(options));
```

This partly resolves #5 – we now create standard Express middleware. You can't access the `request` object in the about endpoint, but it can now be generated asynchronously.

The best way to view any breaking changes is to [read the migration guide](https://github.com/Financial-Times/express-web-service/blob/56d760be6f765cfeb74354da3c2cf8c30f04bfdb/README.md#migration).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,41 +1,97 @@
...
-# Example
+### Options
+
+The available options are as follows:
+
+  - `manifestPath`: (Optional) Path to the app's `package.json` file. This will be used to populate the `appVersion` and `dateDeployed` properties of the `/__about` endpoint, if they are not specified explicitly
+  - `about`: (Optional) Object containing standard runbook properties as defined in the [ft about endpoint standard], or a Promise that resolves to an object containing these properties
+  - `goodToGoTest`: (Optional) A function that can be used to indicate the good to go status of the service, the function should return a Promise resolved with `true` to indicate a positive good to go status, and `false` to indicate a negative good to go status
+  - `healthCheck`: (Optional) A function that can be used to generate structured healthcheck information, the function should return a Promise resolved with an array of healthcheck objects
+  - `routes`: (Optional) An array of routes to install.  Possible values are `health`, `gtg`, `about` and `error`.  Defaults to `["health", "gtg", "about"]`
```
Comment (lucas42) [inline README.md]: I'm not sure this is immediately clear.  I had to read the code before I got what it means.  Perhaps an example would help?

Comment (lucas42) [inline README.md]: I'm not sure this is immediately clear.  I had to read the code before I got what it means.  Perhaps an example would help?

File: lib/express-web-service.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	// Serve the __error endpoint
+	middleware.error = (request, response, next) => {
+		if (!options.routes.includes('error')) {
+			return next();
+		}
+		next(new Error('Intentional test error triggered by the /__error route'));
+	};
+
+	// Serve the __gtg endpoint
+	// TODO
```
Comment (lucas42) [inline lib/express-web-service.js]: What's TODO here?

File: README.md (hunk 1 above)
Comment (rowanmanning) [inline README.md]: Good idea, will add an issue for this

File: lib/express-web-service.js (hunk 2 above)
Comment (rowanmanning) [inline lib/express-web-service.js]: Oops, will remove

**Pull request merged**